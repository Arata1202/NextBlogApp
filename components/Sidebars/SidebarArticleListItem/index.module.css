.list {
  margin-bottom: 20px;
}

.date {
  display: flex;
  justify-content: right;
}

.link {
  display: flex;
  gap: 100px;
}

.image {
  width: 130%;
  height: auto;
  max-width: 130%;
}

.description {
  font-size: 14.5px;
}

.content {
  max-width: 490px;
}

.title {
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
}

@media (max-width: 640px) {
  .link {
    display: block;
  }
  .image {
    width: 100%;
    height: auto;
    margin-bottom: 1rem;
  }

  .title {
    font-size: 1.4rem;
  }
}
/* iPadProç”¨ */
/* @media (min-width: 1024px) and (max-width: 1090px) {
  .image {
    width: 210%;
    height: auto;
    max-width: 210%;
  }
} */
