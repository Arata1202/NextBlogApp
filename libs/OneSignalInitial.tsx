'use client';

import { useEffect } from 'react';
import OneSignal from 'react-onesignal';

export const OneSignalInitial = () => {
  useEffect(() => {
    const oneSignalInit = async () => {
      await OneSignal.init({
        appId: process.env.NEXT_PUBLIC_ONESIGNAL_APP_ID || '',
        notifyButton: {
          enable: true,
        },
      }).then(() => {
        OneSignal.Slidedown.promptPush();
      });
    };
    oneSignalInit();
  }, []);
  return (
    <div className="mt-5 flex justify-center">
      <div className="onesignal-customlink-container"></div>
    </div>
  );
};
