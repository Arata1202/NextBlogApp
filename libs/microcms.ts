import { createClient } from 'microcms-js-sdk';
import type {
  MicroCMSQueries,
  MicroCMSImage,
  MicroCMSDate,
  MicroCMSContentId,
} from 'microcms-js-sdk';
import { notFound } from 'next/navigation';

// カテゴリーの型定義
export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// タグの型定義
export type Tag2 = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// ライターの型定義
export type Writer = {
  name: string;
  profile: string;
  image?: MicroCMSImage;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type ContentBlock = {
  adsense?: string;
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type RelatedArticle = {
  articleLink3?: string;
};

// ブログの型定義
export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  tags?: Tag[];
  tags2?: Tag2[];
  content_blocks: ContentBlock[];
  introduction_blocks: IntroductionBlock[];
  related_articles?: RelatedArticle[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

// Initialize Client SDK.
export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

// ブログ一覧を取得
export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);
  return listData;
};

// ブログの詳細を取得
export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// カテゴリーの一覧を取得
export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

// タグの一覧を取得
export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag2>({
      endpoint: 'tags2',
      queries,
    })
    .catch(notFound);

  return listData;
};

// カテゴリーの詳細を取得
export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// タグの詳細を取得
export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag2>({
      endpoint: 'tags2',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
