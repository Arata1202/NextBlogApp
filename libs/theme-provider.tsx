'use client';

import { ThemeProvider as NextThemesProvider } from 'next-themes';
import type { ThemeProviderProps } from 'next-themes';
import { FC, useState, useEffect } from 'react';

export const ThemeProvider: FC<ThemeProviderProps> = (props) => {
  const [isMounted, setIsMounted] = useState(false);

  useEffect(() => {
    setIsMounted(true);
  }, []);

  if (!isMounted) {
    return null;
  }

  return <NextThemesProvider {...props}>{props.children}</NextThemesProvider>;
};
