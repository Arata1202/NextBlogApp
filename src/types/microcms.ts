import type { MicroCMSImage, MicroCMSDate, MicroCMSContentId } from 'microcms-js-sdk';

export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  categories: Category[];
  tags?: Tag[];
  introduction_blocks: IntroductionBlock[];
  content_blocks: ContentBlock[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

export type Category = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};

export type ContentBlock = {
  google_adsense?: string;
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
