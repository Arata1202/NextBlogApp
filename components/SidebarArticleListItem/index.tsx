'use client';

import React, { memo } from 'react';
import Image from 'next/image';
import { Article } from '@/libs/microcms';

type Props = {
  article: Article;
};

function getImageSrcSizes(imageSrc: string) {
  return {
    mobileSrcSet: `${imageSrc}?fm=webp&w=414 1x, ${imageSrc}?fm=webp&w=414&dpr=2 2x`,
    desktopSrcSet: `${imageSrc}?fm=webp&fit=crop&w=240&h=126 1x, ${imageSrc}?fm=webp&fit=crop&w=240&h=126&dpr=2 2x`,
  };
}

const SidebarArticleListItem = ({ article }: Props) => {
  const imageSrc = article.thumbnail?.url || '/no-image.png';
  const isThumbnailAvailable = !!article.thumbnail;
  const { mobileSrcSet, desktopSrcSet } = getImageSrcSizes(imageSrc);

  const handleClick = (event: React.MouseEvent) => {
    event.preventDefault();
    window.location.href = `/articles/${article.id}`;
  };

  return (
    <div>
      <ol
        className="ArticleListItem_list border mt-5 border-gray-300 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1"
        onClick={handleClick}
      >
        <li>
          <div>
            <Image
              src={imageSrc}
              alt="サムネイル"
              width="800"
              height="450"
              placeholder="blur"
              blurDataURL={imageSrc}
            />
            <div>
              <div className="ArticleListItem_title font-bold">{article.title}</div>
            </div>
          </div>
        </li>
      </ol>
    </div>
  );
};

export default memo(SidebarArticleListItem);
