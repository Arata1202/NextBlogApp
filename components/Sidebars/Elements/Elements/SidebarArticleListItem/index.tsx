'use client';
import { useTheme } from 'next-themes';
import React, { memo } from 'react';
import Image from 'next/image';
import { Article } from '@/libs/microcms';

type Props = {
  article: Article;
};

const SidebarArticleListItem = ({ article }: Props) => {
  const imageSrc = article.thumbnail?.url || '/no-image.png';

  const handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
    window.location.href = `/articles/${article.id}`;
  };

  const { theme } = useTheme();

  return (
    <div>
      <ol
        className={`ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <li>
          <a href={`/articles/${article.id}`} className="sidebarPopularPost" onClick={handleClick}>
            <Image
              src={imageSrc}
              alt="サムネイル"
              width="600"
              height="300"
              placeholder="blur"
              blurDataURL={imageSrc}
            />
            <div>
              <div className="ArticleListItem_title font-bold">{article.title}</div>
            </div>
          </a>
        </li>
      </ol>
    </div>
  );
};

export default memo(SidebarArticleListItem);
