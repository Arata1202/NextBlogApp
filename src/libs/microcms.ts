import { notFound } from 'next/navigation';
import { createClient } from 'microcms-js-sdk';
import type { MicroCMSQueries } from 'microcms-js-sdk';
import { Blog, Category, Tag } from '@/types/microcms';

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

export const getAllLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllCategoryLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Category>({
      endpoint: 'categories',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllTagLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
