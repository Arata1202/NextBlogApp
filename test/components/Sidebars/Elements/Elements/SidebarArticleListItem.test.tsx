import { render } from '@testing-library/react';
import SidebarArticleListItem from '@/components/Sidebars/Elements/Elements/SidebarArticleListItem';
import { Article } from '@/libs/microcms';

const article: Article = {
  id: '1',
  title: 'タイトル',
  description: '概要',
  thumbnail: { url: 'https://thumbnail.com/thumbnail.jpg' },
  publishedAt: '2023-10-30T12:00:00Z',
  content_blocks: [],
  introduction_blocks: [],
  createdAt: '2023-10-30T12:00:00Z',
  updatedAt: '2023-10-30T12:00:00Z',
};

describe('SidebarArticleListItem', () => {
  test('スナップショット（SidebarArticleListItem）', () => {
    const { asFragment } = render(<SidebarArticleListItem key={article.id} article={article} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
