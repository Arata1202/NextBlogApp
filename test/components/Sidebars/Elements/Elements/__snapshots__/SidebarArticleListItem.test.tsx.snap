// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarArticleListItem スナップショット（SidebarArticleListItem） 1`] = `
<DocumentFragment>
  <div>
    <ol
      class="ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 LightTheme"
    >
      <li>
        <a
          class="sidebarPopularPost"
          href="/articles/1"
        >
          <img
            alt="サムネイル"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75 2x"
            style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="600"
          />
          <div>
            <div
              class="ArticleListItem_title font-bold"
            >
              タイトル
            </div>
          </div>
        </a>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
